test/integration/eds_integration_test.cc

```diff
@@ -195,6 +206,27 @@ class EdsIntegrationTest
 
   void initializeTest(bool http_active_hc) { initializeTest(http_active_hc, nullptr); }
 
+  void dropOverloadTest(uint32_t numerator, std::string status) {
+    autonomous_upstream_ = true;
+
+    initializeTest(false);
+    EndpointSettingOptions options;
+    options.total_endpoints = 2;
+    options.healthy_endpoints = 2;
+    options.drop_overload_numerator = numerator;
+    setEndpoints(options);
+
+    // Check deferred.
+    if (deferred_cluster_creation_) {
+      test_server_->waitForGaugeEq("thread_local_cluster_manager.worker_0.clusters_inflated", 0);
+    }
+    BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
+        lookupPort("http"), "GET", "/cluster_0", "", downstream_protocol_, version_, "foo.com");
+    ASSERT_TRUE(response->complete());
```

Comment:
consider validating the access logs since that's not tested elsewhere

Reply:
Please take a look the newly added test: router_test.cc:DropOverloadDropped.